### ISSUE TITLE
unicorn ctl_remove_cache problem (inconsistent between Python 2 and 3?)

### ISSUE THREAD

**Issue by gerph** [unicorn-engine/unicorn#1864]
Hiya,

I'm trying to work out how to use ctl_remove_cache efficiently with Unicorn 2 on Python 2.

I've installed the build from the `restore-py2` branch (this might be wrong, but it seems like where I should be looking), by running 
`python setup.py bdist` and then installing the wheel that's created with `pip install dist/unicorn-2.0.2-py2.py3-none-macosx_10_14_x86_64.whl`.

However, when I look at the `unicorn_py2.py` file, I see that the remove_cache call is defined as:

```
    def ctl_remove_cache(self, addr):
        self.__ctl_w_1_arg(uc.UC_CTL_TB_REMOVE_CACHE, addr, ctypes.c_uint64)
```

This looks like I have to call the `ctl_remove_cache` for every single address that I have modified - that can get very tedious if a large block of code has been loaded and needs to be updated.

In the python 3 version in the source (`unicorn_py3/unicorn.py`), however, I see:

```
    def ctl_remove_cache(self, addr: int, end: int):
        self.__ctl_w_2_arg(uc.UC_CTL_TB_REMOVE_CACHE, addr, end, ctypes.c_uint64, ctypes.c_uint64)
```

This looks like it's a range, which is a lot better for efficiency.

I'm not sure whether the interface is actually different on Python 2, or whether they're inconsistent, or if I'm even using the right branch to try to get Python 2 support to work.

Can you help resolve what I need to do?

**Comment by wtdcode** [comment]
That’s a bug in Py2 and I will push a fix for that.

**Comment by wtdcode** [comment]
I pushed fix at b6cedad79a05dfc8dafb65b086c4332b1627cd97.

But please note that, before 2.1.0 release, the content in `restore-py2` may be changed without notice though the major APIs are guaranteed not changed.

**Comment by wtdcode** [comment]
As fixed already (the patch will go to next release for sure!), I will close this issue.
